apps/api/src/prisma/schema.prisma

```diff
@@ -518,3 +518,16 @@ model ChangeNotificationSocketMap {
 
   @@index([environmentId, socketId])
 }
+
+model UserEmailChange {
+  id            String    @id @default(cuid())
+  userId        String    
+  newEmail      String
+  otp           String
```

Comment:
One thing that you can do is, de-normalize this thing. You see, the `Otp` table already has a `userId` field. What we can do is, instead of re-adding all the fields to `UserEmailChange`, we can just extend the functionality. 

What I mean is, you can add a one-to-one mapping of Otp on UserEmailChange. That way, your schema will look something like this:

```prisma
model Otp {
  code      String   @unique
  user      User     @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId    String   @unique
  createdAt DateTime @default(now())
  expiresAt DateTime

  @@unique([userId, code], name: "userCode")
  @@index([expiresAt], name: "expiresAt")
}

model UserEmailChange {
  id            String    @id @default(cuid())
  otpId       String
  otp          Otp
  newEmail      String
}
```

This would be an easier way to approach this problem, where we are using just one otp table to help our functionality.

Reply:
Done